inventory/sample-ha.aws.example.com.d/inventory/group_vars/all.yml

```diff
@@ -0,0 +1,56 @@
+# 'hosting_infrastructure' is used to drive the correct behavior based
+# on the hosting infrastructure, cloud provider, etc. Valid values are:
+# - 'openstack'
+# - 'aws' (Coming Soon)
+# - 'azure' (Coming Soon)
+hosting_infrastructure: aws
+
+# AWS Provisioning Variables
+# use the "-e" option to set "aws_ssh_public_key"
+aws_image_name: <ec2 AMI>
+aws_num_masters: 3
+aws_num_nodes: 5
+aws_num_infra: 3
+aws_access_key: <Access key ID>
+aws_secret_key: <Secret access key>
+aws_region: <AWS Region>
+aws_key_name: <Key pair name>
+aws_subnet: <Subnet ID>
+# If aws_create_vpc is 'true', aws_subnet variable will be ignored
+aws_create_vpc: <true|false>
+# These are the security groups created when aws_create_vpc is 'true'. Modify accordingly to your environment in case using existing VPC and SGs
```

Comment:
Could we make the role create a set of default security groups and hence don't have to specify these in every inventory?

Reply:
So the idea here is to automatically create everything related to VPC only if 'aws_create_vpc' is 'true'. In other cases, both subnet (will change to subnetS soon) and security groups must be provided.

What is your suggestion here? To create these SGs and not allow the user to select their owns? My point of view is, if the user wants to use existing subnets, he/she will probably want to use existing SGs as he/she has the knowledge to do that.

Reply:
Agreed. My point was to assign default values in the role and hence don't have to require these to be set in the inventory when `aws_create_vpc` is `true`. In many cases I suspect that the users will be ok with using the default values and hence there's no need to require them in the inventory.